src/azul/service/source_controller.py

```diff
@@ -63,3 +68,14 @@ def _add_implicit_sources_filter(self,
         if explicit_source_ids is not None:
             source_ids = source_ids.intersection(explicit_source_ids)
         explicit_filters['sourceId']['is'] = list(source_ids)
+
+    def _parse_filters(self,
+                       filters: Optional[Union[str, Filters]],
+                       source_ids: Optional[Set[str]] = None) -> MutableFilters:
+        if filters is None or isinstance(filters, str):
+            filters = super()._parse_filters(filters)
+        else:
+            filters = copy_json(filters)
+        if source_ids is not None:
+            self._add_implicit_sources_filter(filters, source_ids)
```

Comment:
This has nothing to do with parsing. The fact that the parsing of filters is done in close proximity to the application of the source filter is incidental and shouldn't lead to a conflation of the two operations.